### PR TITLE
Feature mantra datatypes

### PR DESCRIPTION
**Pull request by carlospgraciano**
**Details**

This pull request contains the addition of Mantra Data Types that handle the sanitizing of an entity's fields. The following actions were taken in order to add the data types:

- Created an abstract class for all Mantra Data Types.
- Created a MantraString to sanitize all the fields whose data contains a string.
- Created a MantraNumber to sanitize all the fields whose data contains a number.
- Created a MantraBoolean to sanitize all the fields whose data contains a boolean.
- Created a Factory for all the Mantra Data Types.
- In MantraState abstract class, the "_dataType" property was created.
- In MantraState abstract class, 2 new methods were created to generate a MantraDataType object using the data types factory.

Each class created or modified in this pull request was documented using the JSDoc convention.

### CODE REVIEW

**Review by jonatangerman — APPROVED** [review]

File: Mantra.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { MantraPlugin } from './lib/MantraPlugin';
 import { error } from './lib/utils';
 
 /*
-    This functions are meant to be used as utilities for the params validation
+    These functions are meant to be used as utilities for the params validation
 */
 const isTruly = (value) => (!!value);
```
Comment (SparoHawk) [inline Mantra.js:7]: Just as a comment. Have you considered using libraries for these types of validations? Such as [Lodash](https://lodash.com/docs/4.17.15) or [Ramda](https://ramdajs.com/docs/)?

Comment (jonatangerman) [reply Mantra.js:7]: Thanks for the concern @SparoHawk we have considered [Lodash](https://lodash.com/docs/4.17.15) and actually in future updates we are going to include it as  a dependency 'tree shaking' more 'complex' utilities we are needing such as the **_.get()** and **_.set()** methods.

For simpler utilities such as the ones used so far we rather prefer keeping dependencies at minimum, if it doesn't trigger further overheads.

Comment (SparoHawk) [reply Mantra.js:7]: Remember to use Lodash's tree shaking. So you just import what you need.

`import _get from 'lodash/get'`;

And there also is: [lodash-modularized](https://www.npmjs.com/search?q=keywords:lodash-modularized)

Comment (jonatangerman) [reply Mantra.js:7]: Totally! Thanks for the suggestion!

**Pull request merged**